packages/@lwc/synthetic-shadow/src/faux-shadow/slot.ts

```diff
@@ -107,74 +86,115 @@ function getFilteredSlotFlattenNodes(slot: HTMLElement): Node[] {
     );
 }
 
-interface AssignedNodesOptions {
-    flatten?: boolean;
+export function assignedSlotGetterPatched(this: Element): HTMLSlotElement | null {
+    const parentNode = parentNodeGetter.call(this);
+    /**
+     * if it doesn't have a parent node,
+     * or the parent is not an slot element
+     * or they both belong to the same template (default content)
+     * we should assume that it is not slotted
+     */
+    if (
+        isNull(parentNode) ||
+        !isSlotElement(parentNode) ||
+        getNodeNearestOwnerKey(parentNode) === getNodeNearestOwnerKey(this)
+    ) {
+        return null;
+    }
+    return parentNode as HTMLSlotElement;
 }
 
-export function PatchedSlotElement(elm: HTMLSlotElement): HTMLSlotElementConstructor {
-    const Ctor = PatchedElement(elm) as HTMLSlotElementConstructor;
-    const { addEventListener: superAddEventListener } = elm;
-    return class PatchedHTMLSlotElement extends Ctor {
-        addEventListener(
+defineProperties((window as any).HTMLSlotElement.prototype, {
```

Comment:
Why do we sometimes access the global types off window and sometimes directly? e.g. https://github.com/salesforce/lwc/pull/1322/files#diff-6c6ff95908ccf01c5d2cf5d9a267b4d4R56
I assume it is because of the funky stuff we do with JSDOM for HTMLSlotElement.

Reply:
it is not longer about JSDOM, but it is about typescript... in this case it is fine to remove that... will do...